### PR TITLE
Drill 2908: Int96 support

### PR DESCRIPTION
**Pull request by parthchandra**

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java [hunk 1]
```diff
@@ -59,6 +63,50 @@ protected void readField(long recordsToReadInThisPass) {
...
+        for (int i = 0; i < recordsReadInThisIteration; i++){
+          currDictValToWrite = pageReader.dictionaryValueReader.readBytes();
+          mutator.setSafe(valuesReadInCurrentPass + i, currDictValToWrite.toByteBuffer(), 0,
+              currDictValToWrite.length());
+        }
+        // Set the write Index. The next page that gets read might be a page that does not use dictionary encoding
+        // and we will go into the else condition below. The readField method of the parent class requires the
+        // writer index to be set correctly.
+        int writerIndex = castedVector.getBuffer().writerIndex();
+        castedVector.getBuffer().setIndex(0, writerIndex + (int)readLength);
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: I'm probably missing it, but I couldn't find where `readLength` is set if we don't call `super.readField()`. 
Does it mean writeIndex is unchanged when reading from the dictionary ?

Comment (parthchandra) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: It's set in the parent class NullableColumnReader::processPages()

Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: `NullableColumnReader` is not one of the parent classes of `ParquetFixedWidthDictionaryReaders `

Comment (parthchandra) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: You're right. For DictionaryFixedBinaryReader, the readField code does not call the super.readField method and the readLength is set in the parent class' method. Updated the patch.

**Comment by adeneche** [comment]
+1, LGTM

**Pull request closed without merging**